code/modules/surgery/organs/liver.dm

```diff
@@ -32,17 +32,14 @@
 			var/provide_pain_message = HAS_NO_TOXIN
 			if(filterToxins && !HAS_TRAIT(owner, TRAIT_TOXINLOVER))
 				//handle liver toxin filtration
-				for(var/I in C.reagents.reagent_list)
-					var/datum/reagent/pickedreagent = I
-					if(istype(pickedreagent, /datum/reagent/toxin))
-						var/thisamount = C.reagents.get_reagent_amount(pickedreagent)
-						if (thisamount <= toxTolerance && thisamount)
-							C.reagents.remove_reagent(pickedreagent, 1)
-						else
-							damage += (thisamount*toxLethality)
-							var/datum/reagent/toxin/found_toxin = pickedreagent
-							if(provide_pain_message != HAS_PAINFUL_TOXIN)
-								provide_pain_message = found_toxin.silent_toxin ? HAS_SILENT_TOXIN : HAS_PAINFUL_TOXIN
+				for(var/datum/reagent/toxin/T in C.reagents.reagent_list)
+					var/thisamount = C.reagents.get_reagent_amount(T.type)
+					if (thisamount <= toxTolerance && thisamount)

```

Comment:
shouldn't you flip this? you should check thisamount is not 0 (lol...) before checking to see if it's relative to toxtol